sonar-db/src/main/java/org/sonar/db/version/v53/FixMsSqlCollation.java

```diff
@@ -0,0 +1,553 @@
+import java.util.List;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
+import org.sonar.db.Database;
+import org.sonar.db.dialect.MsSql;
+import org.sonar.db.version.AlterColumnsBuilder;
+import org.sonar.db.version.DdlChange;
+
+import static com.google.common.base.Preconditions.checkState;
+import static java.util.Objects.requireNonNull;
+import static org.sonar.db.version.ColumnDefValidation.validateColumnName;
+import static org.sonar.db.version.TextColumnDef.newTextColumnDefBuilder;
+import static org.sonar.db.version.VarcharColumnDef.newVarcharColumnDefBuilder;
+
+/**
+ * Fix the collation of all columns on MsSQL
+ */
+public class FixMsSqlCollation extends DdlChange {
+
+  private static final Logger LOGGER = Loggers.get(FixMsSqlCollation.class);
+
+  private final Database db;
+
+  public FixMsSqlCollation(Database db) {
+    super(db);
+    this.db = db;
+  }
+
+  @Override
+  public void execute(Context context) throws SQLException {
+    if (!getDatabase().getDialect().getId().equals(MsSql.ID)) {
+      return;
+    }
+
+    // characteristics
+    new UpdateTableCollation(context, db, "characteristics")
+      .addVarcharColumn("kee", 100)
+      .addVarcharColumn("name", 100)
+      .addVarcharColumn("function_key", 100)
+      .addVarcharColumn("factor_unit", 100)
+      .addVarcharColumn("offset_unit", 100)
+      .execute();
+
+    // rules_parameters
+    new UpdateTableCollation(context, db, "rules_parameters")
+      .addNotNullableVarcharColumn("name", 128)
+      .addNotNullableVarcharColumn("param_type", 512)
+      .addVarcharColumn("default_value", 4000)
+      .addVarcharColumn("description", 4000)
+      .execute();
+
+    // rules_profiles
+    new UpdateTableCollation(context, db, "rules_profiles")
+      .addUniqueIndex("uniq_qprof_key", "kee")
+      .addNotNullableVarcharColumn("name", 100)
+      .addVarcharColumn("language", 20)
+      .addNotNullableVarcharColumn("kee", 255)
+      .addVarcharColumn("parent_kee", 255)
+      .addVarcharColumn("rules_updated_at", 100)
+      .execute();
+
+    // project_qprofiles
+    new UpdateTableCollation(context, db, "project_qprofiles")
+      .addUniqueIndex("uniq_project_qprofiles", "project_uuid", "profile_key")
+      .addNotNullableVarcharColumn("project_uuid", 50)
+      .addNotNullableVarcharColumn("profile_key", 255)
+      .execute();
+
+    // widgets
+    new UpdateTableCollation(context, db, "widgets")
+      .addIndex("widgets_widgetkey", "widget_key")
+      .addNotNullableVarcharColumn("widget_key", 256)
+      .addVarcharColumn("name", 256)
+      .addVarcharColumn("description", 1000)
+      .execute();
+
+    // groups
+    new UpdateTableCollation(context, db, "groups")
+      .addVarcharColumn("name", 500)
+      .addVarcharColumn("description", 200)
+      .execute();
+
+    // snapshots
+    new UpdateTableCollation(context, db, "snapshots")
+      .addIndex("snapshots_qualifier", "qualifier")
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "qualifier" 5 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1192)